### PR TITLE
Replace circleci to use github actions

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
some updates:
- replace circleci to use github actions

the ci will run to build/test and build the image but not push more for testing purposes, when we merge to the main branch or create a tag the image will be built and push using the internal gitlab.

the circleci for this repository is already disabled

#### Ticket Link
n/a

### CODE REVIEW

**Comment by cpanato** [comment]
also enabled `Require approval for all outside collaborators`

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    - name: check assets
+      run: |
+        make assets
+        echo "Checking if 'make assets' was ran after adding files..."
+        git diff --exit-code
+
+    - name: check dependencies
+      run: |
+        go mod tidy -v
+        if [[ -n $(git --no-pager diff --exit-code go.mod go.sum) ]]; then echo "There are unused dependencies that should be removed. Please execute `go mod tidy` to fix it."; exit 1; fi
```
Comment (spirosoik) [inline .github/workflows/build.yml:34]: same comment in a the other PR

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Build and push
+      uses: docker/build-push-action@v2
+      with:
+        file: ./Dockerfile
+        push: false
+        tags: not-to-push/mattermod:latest
```
Comment (spirosoik) [inline .github/workflows/build.yml:54]: why do you need the `not-to-push` as you have the `push: false`? Also why not just typical `make build-image`. I am a bit confused here

**Review by spirosoik — COMMENTED** [review]
few questions

File: .github/workflows/build.yml (hunk 2 above)
Comment (cpanato) [reply .github/workflows/build.yml:54]: this is just to check if the docker build works we are not pushing images for prs

File: .github/workflows/build.yml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/build.yml]: done

File: .github/workflows/build.yml (hunk 2 above)
Comment (cpanato) [reply .github/workflows/build.yml]: updated the name
thanks for the review

**Review by spirosoik — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -35,6 +35,14 @@ clean:
 check-style: golangci-lint
 	@echo Checking for style guide compliance
 
+.PHONY: check-deps
+check-deps:
+	$(GO) mod tidy -v
+	@if [ -n "$$(command git --no-pager diff --exit-code go.mod go.sum)" ]; then \
```
Comment (agnivade) [inline Makefile:41]: Is there a need to execute via command?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        image: mysql:5.7
+        env:
+          MYSQL_ROOT_PASSWORD: mattermod
+          MYSQL_DATABASE: mattermod
+          MYSQL_USER: mattermod
+          MYSQL_PASSWORD: mattermod
+        ports:
+        - 3306:3306
+        # needed because the mysql container does not provide a healthcheck
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=10s --health-retries=10
```
Comment (agnivade) [inline .github/workflows/test.yml:30]: Does this ensure the "wait till MySQL is ready"?

Comment (cpanato) [reply .github/workflows/test.yml:30]: during all the running i did to test it was always up, but we can add a check just to make sure

File: Makefile (hunk 3 above)
Comment (cpanato) [reply Makefile:41]: will add this a make rule similar we did for others, this is a good check we have in other repos, and was running here as well, just replicating the same ci again

File: .github/workflows/test.yml (hunk 4 above)
Comment (cpanato) [reply .github/workflows/test.yml:30]: actually, we don't need to do anything gh does all the checks

Screenshot 2022-01-21 at 09 08 36

**Review by agnivade — APPROVED** [review]
Thanks Carlos :+1:

**Pull request merged**